kolibri/core/content/serializers.py

```diff
@@ -438,24 +445,13 @@ class Meta:
 
     @property
     def channel_stats(self):
-        channel_stats = self.context["channel_stats"]
-        if channel_stats is not None:
-            # Convert ContentNode IDs to hex strings if they are UUID
-            # objects
-            stats_convert_id = {}
-            for key in channel_stats.keys():
-                if not isinstance(key, UUID):
-                    return channel_stats
-                contentnode_id = key.hex
-                stats_convert_id[contentnode_id] = channel_stats[key]
-            channel_stats = stats_convert_id
-        return channel_stats
+        return self.context["channel_stats"]
 
     def get_total_resources(self, instance):
         # channel_stats is None for export
         if self.channel_stats is None:
             return instance.on_device_resources
-        return self.channel_stats.get(instance.id, {"total_resources": 0})[
+        return self.channel_stats.get(coerce_key(instance.id), {"total_resources": 0})[
```

Comment:
I tested the PR but one thing I found is that the error indicated in https://github.com/learningequality/kolibri/pull/6544 comes back because in `channel_stats` the ids are stored as UUIDs when connecting to postgresql. But `instance.id` is a string, and `coerce_key(instance.id)` is then also a string.

since we don't want to iterate over `channel_stats` to check that, maybe we can get one item from `channel_stats` to see whether the ids stored inside are UUIDs or strings and then make changes based on that result? - just my two cents though...

please feel free to let me know if there's any debug information you need. thank you!